threedigrid_builder/grid/lines_1d2d.py

```diff
@@ -279,6 +282,24 @@ def assign_2d_node(self, cell_tree: shapely.STRtree) -> None:
         self.line[line_idx, 0] = cell_idx
         self.line_coords[:] = np.nan
 
+    def remove_unassigned(self, nodes) -> None:
+        """Removes 1D-2D lines where any of the required nodes is set to null, represented as -9999
+        This is the case when the nodes are outside the 2D domain.
+        """
+        rows_removed = self[np.where(self.line[:, 0] == -9999)]
```

Comment:
This operation is rather heavy (because it copies over every column); as you only need the node indices, I think you can optimize here a bit. 

And you could also work with bool masks.

```
unassigned = self.line[:, 0] == -9999
node_ids = self.line[unassigned, 1]
```